packages/client/src/debounce.test.tsx

```diff
@@ -0,0 +1,56 @@
+import debounce from "./debounce"
+
+jest.useFakeTimers()
+
+describe("debounce", () => {
+  let func: jest.Mock
+  let debouncedFunc: Function
+
+  beforeEach(() => {
+    func = jest.fn()
+    debouncedFunc = debounce(func, 3000, true)
+  })
+
+  it("executes once over a period shorter than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 300; i++) {
+      debouncedFunc()
+    }
+
+    jest.runAllTimers()
+
+    expect(func).toBeCalledTimes(1)
+  })
+
+  it("calls func every 3 seconds over a period longer than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 10; i++) {
+      debouncedFunc()
+      jest.advanceTimersByTime(3000)
+    }
+
+    expect(func).toBeCalledTimes(10)
```

Comment:
This does not quite test what we want because advancing the timers after every call gives time for `debounce`'s timeout to clear. We want to test that `func` is called every three seconds while `debouncedFunc` is called multiple times during any given three-second window.

I'm calling this out because there is a problem with the implementation of `debounce`. This test and the next test should be adjusted until they both fail.